component/icons.jsx

```diff
@@ -5,26 +5,91 @@ svg.icon {
   height: 2ch;
   margin: -0.5ch 0;
 }
+.top-bar .menu-icon {
+  fill: #fff;
+  margin-right: 2rem;
+  cursor: pointer; }
+  #hamburgerMenu {
+    display: none; }
+    @media only screen and (max-width: 62em) {
+      #hamburgerMenu {
+        display: inline-block; } }
+        .img-logoNaN {
+          height: 3rem; }
 `)
 
-const Icon = (d, props) => (rest) => (
-  <svg class="icon" viewBox="0 0 24 24" {...props} {...rest}>
-    <path d={d} />
-  </svg>
-)
+export const Icon = (title,defs,ds,props) => (rest) => {
+  return <svg version="1.1" xmlns="http://www.w3.org/2000/svg"  {...props} {...rest}>
+    {title && <title>{title}</title>}
+    {defs && <defs>{defs}</defs>}
+    {ds.map((d,i) => <path key={i} d={d}/>)}
+    </svg>
+}
+
 
 export const Github = Icon(
-  'M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22',
-  {
-    'stroke-width': '2',
-    'stroke-linecap': 'round',
-    'stroke-linejoin': 'round',
-    stroke: 'currentColor',
-    fill: 'none',
-  },
+  undefined,
+  undefined,
```

Comment:
that is a terrible API, calling `undefined` to skip arguments is just very bad, this is not a good way to handle optional arguments